test/integration/filters/add_body_filter.cc

```diff
@@ -142,5 +142,9 @@ class AddBodyFilterFactory : public Extensions::HttpFilters::Common::FactoryBase
   }
 };
 
+using UpstreamAddBodyFilterFactory = AddBodyFilterFactory;
+
 REGISTER_FACTORY(AddBodyFilterFactory, Server::Configuration::NamedHttpFilterConfigFactory);
+REGISTER_FACTORY(UpstreamAddBodyFilterFactory,
```

Comment:
nit: for dual factories, perhaps we should have a `REGISTER_DUAL_FACTORY` or similar to enforce both registrations?